packages/nimble-components/src/table/template.ts

```diff
@@ -22,7 +22,7 @@ export const template = html<Table>`
     <template role="table" ${children({ property: 'childItems', filter: elements() })}>
         <div class="table-container">
             <div role="rowgroup" class="header-container">
-                <div class="header-row" role="row">
+                <div class="header-row" role="row" style="grid-template-columns: ${x => x.rowGridColumns} auto; left: -${x => x.scrollX}px;">
```

Comment:
Instead of setting inline styles did we consider other options like configuring these attributes in `styles.ts` to use CSS custom properties and setting the values of those properties from TS code?

My thought is to prefer all styles be configured in CSS files so that it's easier to see the whole picture in one place. It looks like our codebase mostly follows this: if I search files called `template.ts` for `style=`, that pattern is only used in this file. (It's possible this was a conscious decision and I'm forgetting the discussion).

Reply:
So, I believe this would suggest that all of the state we are effectively "binding" styling to would become CSS variables, and the internal implementation that is currently setting observable properties would change to setting these CSS variables in code. Are there other Nimble components that are doing this? Personally, since most (all?) of the styles being set directly here in the template are related to layout changes induced by table state changes, I find it helpful to see that in the template (I think).

Reply:
@jattasNI @rajsite, based on the above conversation, I don't have the best sense of where we want to land. However, since this PR isn't introducing _all_ of the styling bindings in the template, if we do want to change direction, is there value in changing the approach present in this PR for the new style changes?

Reply:
I am more convinced we should move the style logic to `styles.ts` and keep the template focused on setting custom properties that are used by the style logic. I don't have a strong preference between the two approaches in Milan's code samples: the first is less code 👍 but still requires setting the `style` attribute 👎. If you want someone to pick one then I'd probably vote for the first but you'll probably form a more informed opinion by trying it out.

Once we establish a pattern I think we should use it for existing places that we set `style` in this file too. I'm fine with that in this PR or a follow-up.